domain-management/src/main/java/org/jboss/as/domain/management/security/WhoAmIOperation.java

```diff
@@ -85,41 +86,42 @@ private WhoAmIOperation(final Authorizer authorizer) {
     public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
         boolean verbose = VERBOSE.resolveModelAttribute(context, operation).asBoolean();
 
-        Caller caller = context.getCaller();
-        if (caller == null) {
+        SecurityIdentity securityIdentity = context.getSecurityIdentity();
+        if (securityIdentity == null) {
             throw new OperationFailedException(DomainManagementLogger.ROOT_LOGGER.noSecurityContextEstablished());
         }
 
         ModelNode result = context.getResult();
         ModelNode identity = result.get(IDENTITY);
-        identity.get(USERNAME).set(caller.getName());
-        String realm = caller.getRealm();
-        if (realm != null) {
-            identity.get(REALM).set(realm);
-        }
+        identity.get(USERNAME).set(securityIdentity.getPrincipal().getName());
+        // TODO Elytron - We don't currently expose the equivalent of a realm.
+        //String realm = caller.getRealm();
+        //if (realm != null) {
+        //    identity.get(REALM).set(realm);
+        //}
 
         if (verbose) {
-            Set<String> groupSet = caller.getAssociatedGroups();
```

Comment:
No there is no longer an assumption that an identity has groups, an identity just has attributes which may or may not be groups - the attributes are then mapped to roles.